src/psyclone/psyir/nodes/omp_directives.py

```diff
@@ -597,6 +597,142 @@ def update(self):
+
+    :param list children: list of Nodes that are children of this Node.
+    :param parent: the Node in the AST that has this directive as a child.
+    :type parent: :py:class:`psyclone.psyir.nodes.Node`
+    :param grainsize: The grainsize value used to specify the grainsize \
+                      clause on this OpenMP directive. If this is None \
+                      the grainsize clause is not applied. Default \
+                      value is None.
+    :type grainsize: int or None.
+    :param num_tasks: The num_tasks value used to specify the num_tasks \
+                      clause on this OpenMP directive. If this is None \
+                      the num_tasks clause is not applied. Default value \
+                      is None.
+    :type num_tasks: int or None.
+
+    :raises GenerationError: if this OMPTaskloopDirective has both \
+                             a grainsize and num_tasks value \
+                             specified.
+    '''
+    def __init__(self, children=None, parent=None, grainsize=None,
+                 num_tasks=None):
+        self._grainsize = grainsize
+        self._num_tasks = num_tasks
+        if self._grainsize is not None and self._num_tasks is not None:
+            raise GenerationError(
+                "OMPTaskloopDirective must not have both grainsize and "
+                "numtasks clauses specified.")
+        super(OMPTaskloopDirective, self).__init__(children=children,
+                                                   parent=parent)
+
+    @property
+    def dag_name(self):
+        '''
+        :returns: the name to use in the DAG for this node.
+        :rtype: str
+        '''
+        _, position = self._find_position(self.ancestor(Routine))
+        return "OMP_taskloop_" + str(position)
+
+    def node_str(self, colour=True):
+        '''
+        Returns the name of this node with (optional) control codes
+        to generate coloured output in a terminal that supports it.
+
+        :param bool colour: whether or not to include colour control codes.
+
+        :returns: description of this node, possibly coloured.
+        :rtype: str
+        '''
+        return "{0}[OMP taskloop]".format(self.coloured_name(colour))
+
+    def validate_global_constraints(self):
+        '''
+        Perform validation checks that can only be done at code-generation
+        time.
+
+        :raises GenerationError: if this OMPTaskloopDirective is not \
+                                 enclosed within an OpenMP parallel \
+                                 region and an OpenMP single region.
+        '''
+        # It is only at the point of code generation that we can check for
+        # correctness (given that we don't mandate the order that a user
+        # can apply transformations to the code). As an  taskloop
+        # directive, we must have an OMPSerialDirective as an
+        # ancestor back up the tree.
+        if not (self.ancestor(OMPSerialDirective)):
```

Comment:
pylint says the outer parenthesis are unnecessary

Reply:
Hmm, my pylint misses this, which pylint version do you use so I can keep up-to-date (I have 2.9.3 and Python 3.6.9). Have fixed this though.